packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -66,6 +66,30 @@ typedef DragTargetLeave = void Function(Object? data);
 /// Used by [DragTarget.onMove].
 typedef DragTargetMove = void Function(DragTargetDetails<dynamic> details);
 
+/// Signature for the strategy that determines the drag start point.
+///
+/// Used for the injectable [Draggable.dragAnchorStrategy] and
+/// [defaultDragAnchorStrategy]
+typedef DragAnchorStrategy = Offset Function(Draggable<Object> draggable, BuildContext context, Offset position);
+
+/// The default implementation of the strategy to determine the Offset for the
+/// dragStartPoint for the Draggable feedback. When it is [DragAnchor.child] the
+/// RenderBox of the Draggable. When it is [DragAnchor.pointer] it is simply
+/// Offset(0, 0). This strategy can be overridden by passing a function as the
+/// parameter dragAnchorStrategy which follows the [DragAnchorStrategy]
+/// signature.
+Offset defaultDragAnchorStrategy(Draggable<Object> draggable, BuildContext context, Offset position) {
+  switch (draggable.dragAnchor) {
+    case DragAnchor.child:
```

Comment:
This may not be a good implementation if we are going to deprecate dragAnchor in the future.

Ideally we should have two default strategy one for DragAnchor.child another for DragAnchor.pointer, developer can swap them out based on their needs.

Reply:
this was basically the bridge between my approach and the existing approach.

do you have suggestions how the interface for the user of `Draggable` would look like?

some thoughts:

```dart
Draggable(child: Whatever()) // would use DragAnchorStrategies.child like the current default.

Draggable(
  dragAnchorStrategy: DragAnchorStrategies.pointer // to choose the other build in  one
)

Draggable(
  dragAnchorStrategy: (draggable, context, position) {
    // my custom strategy
  }
)
```

what do you think. am i going in the right direction?

Reply:
The original approach uses enum which makes it less customizable. My thinking is this
```dart
Offset defaultPointerStrategy(Draggable<Object> draggable, BuildContext context, Offset position) {
  return Offset.zero;
}

Offset defaultChildStrategy(Draggable<Object> draggable, BuildContext context, Offset position) {
  final RenderBox renderObject = context.findRenderObject()! as RenderBox;
  return renderObject.globalToLocal(position);
}


Draggable(
  dragAnchorStrategy: defaultPointerStrategy // developer can choose which default they want, or provide custom function.
)
```

Reply:
just doing

```dart
Draggable(
  dragAnchorStrategy: defaultPointerStrategy // developer can choose which default they want, or provide custom function.
)
```
wont work. `defaultPointerStrategy` is not a thing on top level or in the scope you are in when using `Draggable` somewhere.

i will implement my thoughts and push later. maybe my example was not complete enough.

Reply:
not sure if i understand, the defaultPointerStrategy can be a static method in Draggable, so if you have access to Draggable, you should be able to use it?

Reply:
> i will implement my thoughts and push later. maybe my example was not complete enough.
played around with it but i am too tired now to finish it.

i would rather not put it as a top level function into `drag_target.dart`. imagine you are building an app and you do 
`import 'package:flutter/material.dart';` and our new function `defaultPointerStrategy` is suddenly in top level namespace. i would not like that.

implementing it as a static method on the `Draggable`. could be a thing, but i would be more explicit and give the strategies their own namespace in the form of a class like:

```dart
class DragAnchorStrategies {
  static DragAnchorStrategy child = (Draggable<Object> draggable, BuildContext context, Offset position) {
    final RenderBox renderObject = context.findRenderObject()! as RenderBox;
    return renderObject.globalToLocal(position);
  };

  static DragAnchorStrategy pointer = (Draggable<Object> _draggable, BuildContext _context, Offset _position) {
    return Offset.zero;
  };
}

```

Reply:
We normally do not create a class just to host static method. I think we should go with static method on draggable. We should follow the pattern we already have in flutter. For example https://api.flutter.dev/flutter/widgets/Navigator/onGenerateInitialRoutes.html

Reply:
also i would like to express that i do not think going for a breaking interface change is a thing that we should consider now. i would suggest to introduce the new, more flexible approach and later deprecate the `dragAnchor`.

i would like to have some more opinions on this.

slightly different topic: in any case this pull request is not really ready. i realized that i did not adapt `LongPressDraggable` yet. i will check out my changes with a real android device to check out how they affect real finger style usage.

Reply:
The reason why i think we should deprecates dragAnchor is that the behavior will be quite confusing if developers try to use both them at the same time. a custom dragAnchorStrategy will override the behavior of dragAnchor entirely. By marking it deprecated, developer can know that dragAnchorStrategy is more preferable than using dragAnchor.

Note that developer can still use deprecated API until we finally remove it after a certain deprecation period, so it is important that we do not change the behavior if developers still use the dragAnchor but not dragAnchorStrategy

Reply:
Ahh. Ok. Understood. During the weekend I will have time again and will look into how to mark things as deprecated in flutter.

Reply:
@chunhtai please have another look. are my newest changes what you are suggesting?

Reply:
while going on with my project that i need this for i just realized that i introduced an error with the last changes. i will investigate.

Reply:
fixed with d940c068830ac22ee0d3c4128c0ca099485ad8a8